apis/v1alpha1/ip_set.go

```diff
@@ -0,0 +1,147 @@
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// IPSetSpec defines the desired state of IPSet.
+//
+// Contains zero or more IP addresses or blocks of IP addresses specified in
+// Classless Inter-Domain Routing (CIDR) notation. WAF supports all IPv4 and
+// IPv6 CIDR ranges except for /0. For information about CIDR notation, see
+// the Wikipedia entry Classless Inter-Domain Routing (https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing).
+//
+// WAF assigns an ARN to each IPSet that you create. To use an IP set in a rule,
+// you provide the ARN to the Rule statement IPSetReferenceStatement.
+type IPSetSpec struct {
+
+	// Contains an array of strings that specifies zero or more IP addresses or
+	// blocks of IP addresses that you want WAF to inspect for in incoming requests.
+	// All addresses must be specified using Classless Inter-Domain Routing (CIDR)
+	// notation. WAF supports all IPv4 and IPv6 CIDR ranges except for /0.
+	//
+	// Example address strings:
+	//
+	//   - For requests that originated from the IP address 192.0.2.44, specify
+	//     192.0.2.44/32.
+	//
+	//   - For requests that originated from IP addresses from 192.0.2.0 to 192.0.2.255,
+	//     specify 192.0.2.0/24.
+	//
+	//   - For requests that originated from the IP address 1111:0000:0000:0000:0000:0000:0000:0111,
+	//     specify 1111:0000:0000:0000:0000:0000:0000:0111/128.
+	//
+	//   - For requests that originated from IP addresses 1111:0000:0000:0000:0000:0000:0000:0000
+	//     to 1111:0000:0000:0000:ffff:ffff:ffff:ffff, specify 1111:0000:0000:0000:0000:0000:0000:0000/64.
+	//
+	// For more information about CIDR notation, see the Wikipedia entry Classless
+	// Inter-Domain Routing (https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing).
+	//
+	// Example JSON Addresses specifications:
+	//
+	//   - Empty array: "Addresses": []
+	//
+	//   - Array with one address: "Addresses": ["192.0.2.44/32"]
+	//
+	//   - Array with three addresses: "Addresses": ["192.0.2.44/32", "192.0.2.0/24",
+	//     "192.0.0.0/16"]
+	//
+	//   - INVALID specification: "Addresses": [""] INVALID
+	//
+	// +kubebuilder:validation:Required
+	Addresses []*string `json:"addresses"`
+	// A description of the IP set that helps with identification.
+	Description *string `json:"description,omitempty"`
+	// The version of the IP addresses, either IPV4 or IPV6.
+	// +kubebuilder:validation:Required
+	IPAddressVersion *string `json:"iPAddressVersion"`
```

Comment:
This is a bug in https://github.com/aws-controllers-k8s/pkg/tree/main/names - needs to be `ipAddressVersion`

Reply:
fixed in https://github.com/aws-controllers-k8s/pkg/pull/31